.ci/jenkins/Jenkinsfile.promote

```diff
@@ -109,36 +105,6 @@ pipeline {
             }
         }
 
-        stage('Merge OptaPlanner Quickstarts PR and tag') {
```

Comment:
I am thinking whether we should rather keep this stage here to have the `development` branch consistent with `main` branches of other repositories.
Only resetting the `stable` branch and all the dependent steps that actually (should) happen from the `stable` branch should be moved outside this file.